packages/hardhat/contracts/FundingVaultFactory.sol

```diff
@@ -99,23 +113,54 @@ contract FundingVaultFactory{
         _withdrawlAddress,
         _developerFeeAddress, 
         _developerFeePercentage, 
-        _projectURL
+        _projectURL,
+        _projectTitle,
+        _projectDescription
         );
 
-        bool tokenTransferSuccess = participationToken.transferFrom(msg.sender, address(fundingVault), _participationTokenAmount);
-        if (!tokenTransferSuccess){
-            revert FundingVault__TokenTransferFailed();
-        }
+        
+        transferParticipationTokens(msg.sender, address(fundingVault), _participationTokenAmount);
 
-        s_fundingVaults[fundingVaultId] = address(fundingVault);
+        Vault storage vault = vaults[fundingVaultId];
+        vault.vaultAddress = address(fundingVault);
+        vault.title = _projectTitle;
+        vault.description = _projectDescription;
+        vault.deadline = _blockLimit;
+        
         emit FundingVaultDeployed(address(fundingVault));     
         return address(fundingVault);
     }
 
+     /**
+     * @notice Get list of all funding vaults
+     * @dev to access the list of all the available funding vaults on the platform 
+     */ 
+    function getVaults(uint256 start, uint256 end) external view returns(Vault[] memory)
+    {
+        if(end > s_fundingVaultIdCounter || start > end || start == 0)
+        {
+            revert FundingVaultFactory__InvalidIndex();
+        }
+        Vault[] memory allVaults = new Vault[](end - start + 1);
+
+        for(uint i = start; i <= end;i++)
+        {
+            allVaults[i - start] = vaults[i];
+        }
+        return allVaults;
+    }
+
+    function transferParticipationTokens(address from, address to, uint256 amount) private{
```

Comment:
We don't need this function. We can call the `transferFrom` function from `participationToken` directly.

Reply:
We had to create a separate function owing to a solidity function stack too deep compiler error..
please suggest alternative solutions if possible.

Reply:
What's wrong with this approach of calling directly?
"participationToken.transferFrom(msg.sender, address(fundingVault), _participationTokenAmount);"

Reply:
I already mentioned that I also did used this approach but this caused stack too deep error. It happens when solidity function is performing too many operations and has lot of local memory.

Reply:
Remove this line uint256 fundingVaultId = s_fundingVaultIdCounter; as it is using a local variable which is really not needed. you can directly use s_fundingVaultIdCounter in Vault storage vault = vaults[s_fundingVaultIdCounter]; 
It's running fine after that.

Reply:
Fixed: Using Safetransfer from Openzepplin resolved this issue. Function removed and calling the safetransferFrom directly in the latest commit